src/include/buffer.h

```diff
@@ -304,7 +310,6 @@ namespace buffer CEPH_BUFFER_API {
 
       void advance(int o);
       void seek(unsigned o);
-      bool operator!=(const iterator_impl& rhs) const;
```

Comment:
jdurgin. the iterator_impl symbols are hidden from librados clients. see https://github.com/ceph/ceph/pull/7974/files#diff-e2d9d51a484aa6d89093b0bcfc70a1e0R271, and `CEPH_BUFFER_DETAILS` is defined as `__attribute__ ((visibility ("hidden")))`.